pkg/container/resolver_test.go

```diff
@@ -104,8 +111,88 @@ func TestResolverFail(t *testing.T) {
 		common.ToPtr(""),
 		common.ToPtr(false),
 		false,
+		nil,
 	})
 	specs, err = resolver.Finish()
 	assert.Error(t, err)
 	assert.Len(t, specs, 0)
 }
+
+func TestResolverLocalManifest(t *testing.T) {
+	currentUser, err := user.Current()
+	if err != nil {
+		t.Fatalf("Failed to get current user: %v", err)
+	}
+
+	if currentUser.Uid != "0" {
+		t.Skip("User is not root, skipping test")
+	}
+
+	_, err = exec.LookPath("podman")
+	if err != nil {
+		t.Skip("Podman not available, skipping test")
+	}
+
+	containerFile, err := os.CreateTemp("", "Containerfile")
+	if err != nil {
+		t.Fatalf("Failed to create tempfile: %v", err)
+	}
+	defer os.Remove(containerFile.Name())
+
+	tmpStorage, err := os.MkdirTemp("", "storage")
+	if err != nil {
+		t.Fatalf("Failed to create temp storage: %v", err)
+	}
+	defer os.Remove(tmpStorage)
+
+	// we need something more than scratch otherwise
+	// two containers with different arches will share
+	// the same underlying container image
+	_, err = containerFile.Write([]byte("FROM fedora-minimal"))
+	if err != nil {
+		t.Fatalf("Failed to create Containerfile: %v", err)
+	}
+
+	cmd := exec.Command( //nolint:gosec
+		"podman",
+		"--root", tmpStorage, // don't dirty the default store
+		"build",
+		"--platform", "linux/amd64,linux/arm64",
+		"--manifest", "multi-arch",
+		"-f", containerFile.Name(),
+		".",
+	)
+
+	err = cmd.Run()
```

Comment:
(niptick) this could have `cmd.Stdout = os.Stdout` and `cmd.Stderr = os.Stderr` here for more visiblity of what is going on during the test or use `output, err := cmd.CombinedOutput()` so that if anything goes wrong there is more debug output.